crates/uv-distribution-types/src/index_url.rs

```diff
@@ -41,6 +41,30 @@ impl IndexUrl {
     ///
     /// Normalizes non-file URLs by removing trailing slashes for consistency.
     pub fn parse(path: &str, root_dir: Option<&Path>) -> Result<Self, IndexUrlError> {
+        Self::parse_with_trailing_slash_policy(path, root_dir, TrailingSlashPolicy::Remove)
+    }
+
+    /// Parse an [`IndexUrl`] from a string, relative to an optional root directory.
+    ///
+    /// If no root directory is provided, relative paths are resolved against the current working
+    /// directory.
+    ///
+    /// Preserves trailing slash if present in `path`.
+    pub fn parse_preserving_trailing_slash(path: &str) -> Result<Self, IndexUrlError> {
```

Comment:
I don't really like this name. Originally I had it as `parse_without_slash_normalization`. Could be `parse_preserve_trailing_slash` but I think `parse_preserve_slash` isn't enough information?

Reply:
I avoided things like `parse_as_given` or `parse_exact` since there could have already been different kinds of normalization prior to this point?

Reply:
What if we just use:

- `parse_simple_api`
- `parse_find_links`

And thus not require callers to pass in a trailing slash policy?

Reply:
Updated